java/src/main/java/ai/rapids/cudf/JCudfSerialization.java

```diff
@@ -598,9 +606,13 @@ private static long getConcatedSerializedDataSizeInBytes(int numColumns, long[]
    * @param numRows the number of rows to serialize.
    * @return the size in bytes needed to serialize the data including the header.
    */
-  public static long getSerializedSizeInBytes(ColumnVector[] columns, long rowOffset, long numRows) {
-    ColumnBufferProvider[] providers = providersFrom(columns);
-    return getSlicedSerializedDataSizeInBytes(providers, rowOffset, numRows) + (4 * 3) + 2; // The header size
+  public static long getSerializedSizeInBytes(HostColumnVector[] columns, long rowOffset, long numRows) {
+    ColumnBufferProvider[] providers = providersFrom(columns, false);
+    try {
+      return getSlicedSerializedDataSizeInBytes(providers, rowOffset, numRows) + (4 * 3) + 2; // The header size
+    } finally {
+      closeAll(providers);
```

Comment:
Curious, is this necessary?  Looks like the code explicitly setup the column providers to not actually close, but then also  goes out of its way to make sure the no-op close methods get called.

Reply:
I personally prefer consistency, even if the code is slightly unnecessary. This way all code that produces a provider will close that provider.  That is the contract with something that is AutoCloseable.